app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt

```diff
@@ -75,13 +77,13 @@ class PatientSearchResultsScreen(context: Context, attrs: AttributeSet) : Relati
     return searchResultsView
         .upstreamUiEvents
         .ofType<RegisterNewPatient>()
-        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.searchPatientInput)) }
+        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.criteria)) }
   }
 
-  private fun extractPatientName(searchPatientInput: SearchPatientInput): String {
-    return when (searchPatientInput) {
-      is SearchPatientInput.Name -> searchPatientInput.searchText
-      is SearchPatientInput.PhoneNumber -> TODO("not yet implemented")
+  private fun extractPatientName(criteria: PatientSearchCriteria): String {
+    return when (criteria) {
+      is Name -> criteria.patientName
+      is PhoneNumber -> TODO("not yet implemented")
```

Comment:
Please implement this.

Reply:
Whoops, good catch! This will be fixed when https://www.pivotaltracker.com/story/show/167343744 is picked up. For now, I will return an empty string so that the app does not crash.